src/custom-element.js

```diff
@@ -343,1087 +343,1100 @@ function isInternalOrServiceNode(node) {
   return false;
 }
 
-
-/**
- * The interface that is implemented by all custom elements in the AMP
- * namespace.
- * @interface
- */
-class AmpElement {
-  // TODO(dvoytenko): Add all exposed methods.
-}
-
 /**
  * Creates a new custom element class prototype.
  *
  * Visible for testing only.
  *
- * @param {!Window} win The window in which to register the elements.
- * @param {string} name Name of the custom element
+ * @param {!Window} win The window in which to register the custom element.
+ * @param {string} name The name of the custom element.
  * @param {function(new:./base-element.BaseElement, !Element)=} opt_implementationClass For
  *     testing only.
  * @return {!Object} Prototype of element.
  */
 export function createAmpElementProto(win, name, opt_implementationClass) {
-  const ElementProto = win.Object.create(createBaseAmpElementProto(win));
-  ElementProto.name = name;
+  const ElementProto = createBaseCustomElementClass(win, name).prototype;
   if (getMode().test && opt_implementationClass) {
     ElementProto.implementationClassForTesting = opt_implementationClass;
   }
   return ElementProto;
 }
 
 /**
- * Creates a new custom element class prototype.
- *
- * The prototype is cached per window and meant to be sub classed for a
- * concrete object.
+ * Creates a new custom element class.
  *
- * @param {!Window} win The window in which to register the elements.
- * @return {!Object} Prototype of element.
+ * @param {!Window} win The window in which to register the custom element.
+ * @param {string} name The name of the custom element.
+ * @return {Function} The custom element class.
```

Comment:
we can probably switch this to `{function(new: BaseCustomElement)}` (not 100% sure)